protocol/packages/dex/src/swap.rs

```diff
@@ -21,9 +21,7 @@ pub trait ExactAmountIn {
         GIn: Group,
         GSwap: Group;
 
-    fn parse_response<I>(trx_resps: &mut I) -> Result<Amount>
-    where
-        I: Iterator<Item = Any>;
+    fn parse_response(response: CosmosAny) -> Result<Amount>;
```

Comment:
do not buy this change without any motivation. Two reasons to reject this change:
- the parse is designed to be reflective of the build fn - the latter allows multiple messages to be sent as a transaction so parse should also be able to parse multiple responses
 - the API allowed communication to more complex DEX APIs

Reply:
The old `parse_response` followed a bad practice, taking the iterator and polling it, which **is** defined as implementation dependent what it will do after the first `None`. This leads to error-proneness.

Another point I'd like to make is that, the iterator iterates over **multiple** responses, while the the function operates on the next *one*.